service/share/discovery.go

```diff
@@ -87,13 +106,25 @@ func (d *discovery) findPeers(ctx context.Context) {
 			for peer := range peers {
 				go d.handlePeerFound(ctx, topic, peer)
 			}
+		case e := <-sub.Out():
+			// listen to disconnect event to remove peer from set and reset backoff time
+			// reset timer in order to restart the discovery, once stored peer is disconnected
+			connStatus := e.(event.EvtPeerConnectednessChanged)
```

Comment:
could *technically* cause panic

```suggestion
			connStatus, ok = e.(event.EvtPeerConnectednessChanged)
			if !ok {
			 // err and return
			}
```

Reply:
It might look like it does, but it fact that channel always returns a type to which it was subscribed

Reply:
if we subscribes on` event.EvtPeerConnectednessChanged` then only this event is expected

Reply:
Yes I understand this, which is why I said "technically". 
However, we have other parts of the codebase that also follow the same pattern, for example https://github.com/celestiaorg/celestia-node/blob/main/core/fetcher.go#L147

In this case too, it is not *technically* possible to receive an event of a diff type via that subscription but we still perform the check

We should be consistent here too.

Reply:
I would prefer to be consistent with things that make sense, like here. We can make GFI to remove the if in the fetcher.